### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by antoniusfilian**
- update user routing

- update admin routing

- login connect to backend

### CODE REVIEW

File: src/app/pages/login-scene/login-scene.component.ts [hunk 1]
```diff
@@ -7,9 +10,40 @@ import { Component, OnInit } from '@angular/core';
...
+      this.theData = data;
+      if (this.theData.success === true) {
+        localStorage.setItem('userToken', this.theData.token);
+        this.loginService.setLoginData(this.theData);
+        this.router.navigateByUrl('home');
+      } else {
+        alert('Wrong Password or Username, Please Try Again!');
+      }
+    },
+    error  => {
```
Comment (sherlyflorencia) [inline src/app/pages/login-scene/login-scene.component.ts]: please delete this console log

File: src/app/pages/login-scene/login-scene.component.ts [hunk 2]
```diff
@@ -7,9 +10,40 @@ import { Component, OnInit } from '@angular/core';
...
+      'password' : this.password
+    })
+    .subscribe(
+      data  => {
+      this.theData = data;
+      if (this.theData.success === true) {
+        localStorage.setItem('userToken', this.theData.token);
+        this.loginService.setLoginData(this.theData);
+        this.router.navigateByUrl('home');
+      } else {
```
Comment (sherlyflorencia) [inline src/app/pages/login-scene/login-scene.component.ts]: Please delete this alert

**Comment by antoniusfilian** [comment]
resolve changes : 

- no console.log

- no alert

update login : 

- spinner loading
image

- empty username or password
image

- wrong username or password
image

File: src/app/app.component.ts [hunk 3]
```diff
@@ -600,7 +600,239 @@ export class AppComponent {
          {name: 'Rezkhi'}
      ]
    }
- ];
+  ];
+  @Output() homeData =
```
Comment (danielindracahyadi) [inline src/app/app.component.ts]: nit: give comment above this output, like so:
// TODO: remove this dummy after connected to Home endpoint

File: src/app/pages/login-scene/login-scene.component.ts [hunk 4]
```diff
@@ -7,9 +10,54 @@ import { Component, OnInit } from '@angular/core';
...
 
-  constructor() { }
+  constructor(
+    private router: Router,
+    private httpClient: HttpClient,
+    private loginService: LoginSceneService,
+  ) { }
+
+  username = '';
+  password = '';
```
Comment (danielindracahyadi) [inline src/app/pages/login-scene/login-scene.component.ts]: nit: give string type

File: src/app/pages/login-scene/login-scene.component.ts [hunk 5]
```diff
@@ -7,9 +10,54 @@ import { Component, OnInit } from '@angular/core';
...
+  empty: boolean = false;
+  errorLogin: boolean = false;
+
+  theData: any;
 
   ngOnInit() {
   }
 
+  login(){
+    this.loading = false;
```
Comment (danielindracahyadi) [inline src/app/pages/login-scene/login-scene.component.ts]: suggestions: desctructure all of this.something below login(){ like this:
let {loading, empty, errorLogin, etc} = this;
next time you won't need to call "this." anymore

File: src/app/pages/login-scene/login-scene.module.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import { TextViewComponent } from 'src/app/core-ui/text-view/text-view.component';
+import { TextInputComponent } from 'src/app/core-ui/text-input/text-input.component';
+import { ButtonComponent } from 'src/app/core-ui/button/button.component';
+
+@NgModule({
+  declarations: [
+    // LoginSceneComponent,
+    // TextViewComponent,
+    // TextInputComponent,
+    // ButtonComponent
```
Comment (danielindracahyadi) [inline src/app/pages/login-scene/login-scene.module.ts]: remove if this isn't used anymore

**Pull request merged**